lib/puppet/parser/scope.rb

```diff
@@ -460,6 +460,17 @@ def lookupdefaults(type)
     values
   end
 
+  # Check if the given value is a known default for the given type
+  #
+  def is_default?(type, key, value)
+    defaults_for_type = @defaults[type]
+    unless defaults_for_type.nil?
+      default_param = defaults_for_type[key]
+      return true if !default_param.nil? && value == default_param.value
```

Comment:
Does this mean that if a value is first set with a default (for example 42), and then overridden with the same value (42), and then overridden again - that it then thinks that the value is default the second time (it isn't since it was explicitly set)?